### ISSUE TITLE
Run load tests for all methods.

### ISSUE THREAD

**Issue by burntcarrot** [burntcarrot/yakv#5]
Currently, yakv has been load tested for PUT requests only. ([Using vegeta with a rate of 50 requests/second](https://github.com/burntcarrot/yakv#5000-put-requests-in-100-seconds-rate--50-requestssecond))

I welcome any pull request to load test for all 3 available methods to test for **maximum possible requests that can be handled by yakv**:

- [x] `GET`
- [ ] `PUT`
- [x] `DELETE`

All load tests can be done using [vegeta](https://github.com/tsenart/vegeta).
For beginners, here's a guide: [Load testing with vegeta](https://www.scaleway.com/en/docs/vegeta-load-testing/)

The load test reports need to have [this specific body](https://github.com/burntcarrot/yakv/blob/main/CONTRIBUTING.md#benchmarking).

**Comment by burntcarrot** [comment]
This issue is open for Hacktoberfest 2021. Feel free to contribute!

**Comment by SkandaSrinivasan** [comment]
Hiya @burntcarrot , I would like to take a shot at this.
Benchmarks for GET:
- Device Specifications:
    - Windows 10 Home 10.0.19043
    - x64
    - AMD Ryzen 7 4700U with Radeon Graphics, 2000 Mhz, 8 Core(s)
    - 16.0 GB
- 5,000 GET requests in 100 seconds (rate = 50 requests/second)
- Available RAM while performing benchmark: 6.7 GB

- Vegeta Report
 ```
Requests      [total, rate, throughput]  5000, 50.01, 50.01
Duration      [total, attack, wait]      1m39.9853379s, 1m39.9847987s, 539.2µs
Latencies     [mean, 50, 95, 99, max]    950.9µs, 1.053607ms, 1.519657ms, 1.762916ms, 3.3781ms
Bytes In      [total, mean]              60000, 12.00
Bytes Out     [total, mean]              75000, 15.00
Success       [ratio]                    100.00%
Status Codes  [code:count]               200:5000
Error Set:
```
- Vegeta Plot
![vegeta-plot](https://user-images.githubusercontent.com/63913769/135390967-44ee1e26-ab19-4424-820e-e4f59739d112.png)

Let me know if I need to change anything or anything is wrong.
 If this looks good, let me know how you want this added to the repo, thanks.

**Comment by burntcarrot** [comment]
Looks good! Just as a reminder, Hacktoberfest starts in less than 24 hours, so you can wait before making a PR. I would happily accept the PR.

Could you please test for a more request/seconds ratio, for example, 20,000 requests in 100 seconds, or somewhat similar, as your device specifications can easily handle more allocations, so I'm wondering whether it could take more load or not.

Feel free to experiment with different ratios, as this issue is for testing the maximum possible load that yakv can take without erroring. Just for some info, you will see either error messages printed to your screen, or you can inspect the chart after the test is done, and if you see any red lines (indicating errors), change the ratio to have a lesser load, and load test again.

Once you're done testing with different ratios, you can add the info to the README (after I give the approval), and make a pull request. After reviewing, I will merge the PR with the `hacktoberfest-accepted` label.

**Comment by burntcarrot** [comment]
@SkandaSrinivasan just for an update: Hacktoberfest has started. You might want to work on this issue soon, as I have multiple people showing interest to work on this issue.

**Comment by SkandaSrinivasan** [comment]
Hi @burntcarrot  thanks for letting me know.

I was able to find a breaking point for yakv today(for GET requests atleast)
```
Requests      [total, rate, throughput]  729995, 7300.56, 6775.31
Duration      [total, attack, wait]      1m41.8330686s, 1m39.991609s, 1.8414596s
Latencies     [mean, 50, 95, 99, max]    745.6489ms, 6.19243ms, 3.373811862s, 4.759803873s, 11.0774477s
Bytes In      [total, mean]              8279412, 11.34
Bytes Out     [total, mean]              10349265, 14.18
Success       [ratio]                    94.51%
Status Codes  [code:count]               0:40044  200:689951
Error Set:
Get "http://127.0.0.1:8880/yakv/v0/get": dial tcp 0.0.0.0:0->127.0.0.1:8880: bind: An operation on a socket could not be performed because the system lacked sufficient buffer space or because a queue was full.
```

7000 req/sec for 100s for a total of 700000 requests, is the last load where there is a 100% success ratio.
Anything above 7000 shows increasing levels of failed requests. Just adding a sample load test for a failing ratio.

Let me know if this is sufficient for a benchmark for GET requests :)

**Comment by burntcarrot** [comment]
Great work! Now you can make the PR.

Make sure to read the [contribution guide](https://github.com/burntcarrot/yakv/blob/main/CONTRIBUTING.md#make-a-pull-request), and make the PR as suggested.

Just add the 50 requests/second portion with your report in README.md, that is, 7000 req/sec for 100s.

Here is the exact part where you should be making your changes: [README.md](https://github.com/burntcarrot/yakv#benchmarks)

**Comment by SkandaSrinivasan** [comment]
Hi @burntcarrot I created a PR and ran the tests and linted my fork beforehand as well just in case. Let me know if i need to make any changes in the PR with regards to formatting or such.

**Comment by burntcarrot** [comment]
Made some suggestions. Can you please look into it?

**Comment by SkandaSrinivasan** [comment]
I'm trying to see any comments you have made on the PR #8 but I cant seem to find any @burntcarrot . Sorry, am i missing something?

**Comment by burntcarrot** [comment]
I added the comment in PR, can you please check again?

**Comment by SkandaSrinivasan** [comment]
@burntcarrot Thanks for all your help. This was fun and informative 💯

**Comment by burntcarrot** [comment]
@SkandaSrinivasan would you like to add benchmarks for `PUT` and `DELETE` requests too? I'd be happy to guide you with the PRs! 😄

**Comment by SkandaSrinivasan** [comment]
@burntcarrot Yup i would love to. I am already getting started on those as well 👍🏽

**Comment by burntcarrot** [comment]
Nice! I'll be waiting for your PRs! 😄

**Comment by burntcarrot** [comment]
Any updates @SkandaSrinivasan? Would you like someone else to work on this issue, if you're not planning to work on this?

**Comment by SkandaSrinivasan** [comment]
hey @burntcarrot  I added the loadtests for PUT.  Looks like delete has been done too , so if you are ok with the the PR we will be done with the loadtests!!

**Comment by burntcarrot** [comment]
@SkandaSrinivasan yes, I'm about to merge it; some minor changes requested.

**Comment by SkandaSrinivasan** [comment]
Hi @burntcarrot I have done the needful.  You can let me know if any further changes are needed :)

**Comment by burntcarrot** [comment]
Resolved via #9.
